### PR TITLE
BasicText - easy-to-use text items to add to Subnautica screen

### PR DESCRIPTION
**Pull request by Cattlesquat**
Super-simple text items (literally 2 lines to get a basic one going -- see main comments for examples). Default is to "clone" the SN in-game text so that you get attractive-looking and SN-suitable text by default w/o having to figure out e.g. what Material to use and what font and etc, etc.  BUT you can set those things too if you want, and it supports "fade out after X seconds".

I don't know how to make the comments the way it seems to want me to (I come from a JavaDoc world and formatted them that way). But anyway have a look.

I have learned SN engine would be more happy if we set layer to "31" (its highest layer) than any other value, though that DOES mean it wouldn't appear on top of the "black screen" when you die. Of course most users probably don't need that anyway.

### CODE REVIEW

File: SMLHelper/Utility/BasicText.cs [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+        protected TextAnchor align { get; set; }      // text alignment
+        protected Color color { get; set; }           // text color
+        protected int size { get; set; }              // text size
+        protected Font font { get; set; }             // text font
+        protected FontStyle style { get; set; }       // text font style
+        protected Material material { get; set; }     // text material
+        protected GameObject textObject { get; set; } = null;          // Our game object
+        protected uGUI_TextFade textFade { get; set; } = null;         // Our text fader
+        protected Text textText { get; set; } = null;                  // Our text object
+        protected ContentSizeFitter textFitter { get; set; } = null;   // Our content size fitter
```
Comment (PrimeSonic) [inline SMLHelper/Utility/BasicText.cs:431]: Is there expectation that we will be extending/inheriting the `BasicText` class?

If not, these could be made `internal`

File: SMLHelper/Utility/BasicText.cs [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+
+        static int index = 0; // For giving unique names to the game objects
```
Comment (PrimeSonic) [inline SMLHelper/Utility/BasicText.cs:433]: If this is to be hidden from public use, make it `internal` explicitly.

File: SMLHelper/Utility/BasicText.cs [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+ * ... // other things happen, time goes by
+ * message.Hide(); // Hides the message
+ */
+using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace SMLHelper.V2.Utility
+{
+    public class BasicText
```
Comment (PrimeSonic) [inline SMLHelper/Utility/BasicText.cs:32]: TODO: Add XML documentation for Visual Studio.
Simply start typing three forward slashes `///` above any member definition and Visual Studio should auto-complete all the necessary XML tags.

File: SMLHelper/Utility/BasicText.cs (hunk 1 above)
Comment (Cattlesquat) [reply SMLHelper/Utility/BasicText.cs]: Made them internal

File: SMLHelper/Utility/BasicText.cs (hunk 2 above)
Comment (Cattlesquat) [reply SMLHelper/Utility/BasicText.cs]: Made it internal

File: SMLHelper/Utility/BasicText.cs (hunk 3 above)
Comment (Cattlesquat) [reply SMLHelper/Utility/BasicText.cs:32]: I tried what you say in Visual Studio, but nothing happened. I must not have something set up right, but Visual Studio is of course a maze of options.

Comment (PrimeSonic) [reply SMLHelper/Utility/BasicText.cs:32]: I forgot that's actually not built-in.
Here's the tool that enables that.
https://submain.com/download/ghostdoc/community/

Comment (Cattlesquat) [reply SMLHelper/Utility/BasicText.cs:32]: image
I have Visual Studio 2019 :O

**Pull request merged**